lib/internal/assert.js

```diff
@@ -191,29 +191,42 @@ function createErrDiff(actual, expected, operator) {
       res += `\n${green}+${white} ${actualLines[i]}`;
       printedLines++;
     // Lines diverge
-    } else if (actualLines[i] !== expectedLines[i]) {
-      if (cur > 1 && i > 2) {
-        if (cur > 4) {
-          res += `\n${blue}...${white}`;
-          skipped = true;
-        } else if (cur > 3) {
-          res += `\n  ${actualLines[i - 2]}`;
+    } else {
+      const expectedLine = expectedLines[i];
+      let actualLine = actualLines[i];
+      let divergingLines = actualLine !== expectedLine &&
+                           (!actualLine.endsWith(',') ||
+                            actualLine.slice(0, -1) !== expectedLine);
+      if (divergingLines &&
+          expectedLine.endsWith(',') &&
+          expectedLine.slice(0, -1) === actualLine) {
+        divergingLines = false;
+        actualLine += ',';
+      }
+      if (divergingLines) {
+        if (cur > 1 && i > 2) {
+          if (cur > 4) {
+            res += `\n${blue}...${white}`;
+            skipped = true;
+          } else if (cur > 3) {
+            res += `\n  ${actualLines[i - 2]}`;
+            printedLines++;
+          }
+          res += `\n  ${actualLines[i - 1]}`;
+          printedLines++;
+        }
```

Comment:
Could you maybe add some comments that match up the different conditionals to the input cases they correspond to…? This is a bit hard to keep track of :smile:

Reply:
I added multiple comments and hope it is clearer now :)

Reply:
It very much is, thank you! :)